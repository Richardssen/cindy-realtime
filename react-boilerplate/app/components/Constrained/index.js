import React from 'react';
import PropTypes from 'prop-types';
import { Box, Flex } from 'rebass';

export function Constrained(props) {
  const { children, level = 2, ...others } = props;
  const styles = {
    flexWrap: 'wrap',
    ml: -2,
    pl: 2,
    justifyContent: 'center',
    alignItems: 'center',
    ...others,
  };
  let MW;
  if (level === 1) {
    MW = [5 / 6, 4 / 5, 3 / 4, 2 / 3];
  } else if (level === 2) {
    MW = [7 / 8, 5 / 6, 4 / 5, 3 / 4];
  } else if (level === 3) {
    MW = [11 / 12, 7 / 8, 5 / 6, 4 / 5];
  } else if (level === 4) {
    MW = [17 / 18, 11 / 12, 7 / 8, 5 / 6];
  } else {
    MW = [1, 17 / 18, 11 / 12, 7 / 8];
  }
  return (
    <Flex {...styles}>
      <Box width={MW}>{children}</Box>
    </Flex>
  );
}

Constrained.propTypes = {
  level: PropTypes.number,
  children: PropTypes.node.isRequired,
};

export default Constrained;
