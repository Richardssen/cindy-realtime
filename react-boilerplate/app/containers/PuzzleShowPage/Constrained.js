import React from 'react';
import PropTypes from 'prop-types';
import { Box, Flex } from 'rebass';

export function Constrained(props) {
  const { children, level = 2, ...others } = props;
  const styles = {
    wrap: true,
    ml: -2,
    pl: 2,
    justify: 'center',
    align: 'center',
    ...others,
  };
  let PW;
  let MW;
  if (level === 1) {
    PW = [1 / 12, 1 / 10, 1 / 8, 1 / 6];
    MW = [5 / 6, 4 / 5, 3 / 4, 2 / 3];
  } else if (level === 2) {
    PW = [1 / 16, 1 / 12, 1 / 10, 1 / 8];
    MW = [7 / 8, 5 / 6, 4 / 5, 3 / 4];
  } else if (level === 3) {
    PW = [1 / 24, 1 / 16, 1 / 12, 1 / 10];
    MW = [11 / 12, 7 / 8, 5 / 6, 4 / 5];
  } else {
    PW = [1 / 36, 1 / 24, 1 / 16, 1 / 12];
    MW = [17 / 18, 11 / 12, 7 / 8, 5 / 6];
  }
  return (
    <Flex {...styles}>
      <Box width={PW} />
      <Box width={MW}>{children}</Box>
      <Box width={PW} />
    </Flex>
  );
}

Constrained.propTypes = {
  level: PropTypes.number,
  children: PropTypes.node.isRequired,
};

export default Constrained;
