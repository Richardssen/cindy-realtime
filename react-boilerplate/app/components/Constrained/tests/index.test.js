import React from 'react';
import { shallow } from 'enzyme';

import Constrained from '../index';

describe('<Constrained />', () => {
  const level = 3;
  it('Expect to render children properly with level 1 ~ 5', () => {
    const wrapper = shallow(
      <Constrained level={level}>
        <div className="test_class" />
      </Constrained>,
    );
    expect(wrapper.contains(<div className="test_class" />)).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });
});
