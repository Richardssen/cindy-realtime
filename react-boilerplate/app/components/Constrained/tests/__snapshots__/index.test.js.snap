// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Constrained /> Expect to render children properly with level 1 ~ 5 1`] = `
<Styled(Flex)
  align="center"
  justify="center"
  ml={-2}
  pl={2}
  wrap={true}
>
  <Styled(Box)
    width={
      Array [
        0.9166666666666666,
        0.875,
        0.8333333333333334,
        0.8,
      ]
    }
  >
    <div
      className="test_class"
    />
  </Styled(Box)>
</Styled(Flex)>
`;
